alacritty/src/display/hint.rs

```diff
@@ -378,7 +381,31 @@ impl<'a, T> HintPostProcessor<'a, T> {
             }
         }
 
-        start..=iter.point()
+        if start > iter.point() {
+            None
+        } else {
+            Some(start..=iter.point())
+        }
+    }
+
+    /// Loops over processed matches until a match is found or the
+    /// range is exhausted.
+    fn iterate_to_next_processed_match(&mut self, mut regex_match: Match) {
+        self.next_match = None;
+
+        while regex_match.start() <= regex_match.end() {
+            self.next_match = self.hint_post_processing(&regex_match);
+
+            regex_match = match self.next_match {
+                Some(_) => break,
+                None => regex_match.start().add(self.term, Boundary::Grid, 1)..=*regex_match.end(),
+            }
+        }
+
+        self.start = self
+            .next_match
+            .as_ref()
+            .map_or(*regex_match.start(), |nm| nm.end().add(self.term, Boundary::Grid, 1));
```

Comment:
Why do you use `regex_match.start()` as fallback here? Because this means that we've mutated `regex_start` to actually be bigger than `regex_match.end()` and we now have to use the end instead of the start to determine the farthest point? Seems a bit confusing.

Reply:
I don't believe I understand the problem with using `regex_match.start()` as the fallback. Isn't it expected that `start` eventually moves past `end` as part of the Iterator implementation?

Reply:
The problem is that it relies heavily on the other code putting you in a state that you don't really want to be in to begin with.